### DISCUSSION TITLE
8-vault-gaurdian-audit

### DISCUSSION THREAD

**Discussion by RishabhTibrewal** [Cyfrin/security-and-auditing-full-course-s23#235] [General]
i wan to know an dverify my finding for the vault-gaurdian smart contracts. where I can look for it. Help me with the github repo or anything please

**Comment by EngrPips** [comment]
Here https://github.com/Cyfrin/8-vault-guardians-audit @RishabhTibrewal

**Comment by RishabhTibrewal** [comment]
the audit-data folder in this repo belongs to the known issues Right!
what about the new findings @EngrPips

**Reply by EngrPips** [reply]
How do you mean the `new findings`?

**Reply by viniciuscoassis** [reply]
I would like to know that too, the audit-data is supposed to have the issues the protocol knew beforehand, but how about the official one ?

**Reply by EngrPips** [reply]
You can maybe make a pull request to add any new finding I think.

**Reply by viniciuscoassis** [reply]
So there's no answer key then ? all the functionalities about vaults, stake values, guardian fees, there was no more bugs than those on audit-data?

**Reply by viniciuscoassis** [reply]
I just rewatched the last video of security part 1 (https://updraft.cyfrin.io/courses/security/mev-and-governance/end-part-1) and Patrick says that he planted a lot of bugs there for us to find and increase our "toolbox" of known issues. And in audit-data theres only like, four bugs, so I must be missing something.

**Reply by EngrPips** [reply]
You can give the codebase one more pass

**Reply by EngrPips** [reply]
> So there's no answer key then ? all the functionalities about vaults, stake values, guardian fees, there was no more bugs than those on audit-data?

I don't understand what you mean.

**Comment by viniciuscoassis** [comment]
I spent a few days working on the Vault Guardians codebase independently, exploring it on my own to identify any potential bugs, without any guidance. I approached it this way to simulate a real public contest, like a Code Hawks challenge. Now that I’ve finished and documented my findings, which I believe are valid, I’d like to compare them to the official answers to understand any mistakes or gaps. Without access to this comparison, I feel I won’t fully benefit from the experience, as I won’t know where I could have improved.

**Reply by EngrPips** [reply]
The `audit data contains the bugs that Patrick planted in the codebase. If you find a bug that is not in the `audit data`, then submit a PR to add the bug to the `audit data`. If the PR is accepted, then your bug is valid, and you can be confident about it.
